src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -253,14 +256,61 @@ public RecurringSession withStartDate(SessionDate newSessionDate) {
     }
 
     /**
-     * Creates a new {@RecurringSession} with the new {@code newLastSessionDate}.
+     * Creates a new {@code RecurringSession} with the new {@code newLastSessionDate}.
      * @param newLastSessionDate the new last session date.
      */
     public RecurringSession withLastSessionDate(SessionDate newLastSessionDate) {
         return new RecurringSession(getSessionDate(), getDuration(), getSubject(),
                 getFee(), getInterval(), newLastSessionDate);
     }
 
+    /**
+     * Removes a single {@code Session} with {@code sessionDate}.
+     * Splits the remaining {@code RecurringSession} into two {@code RecurringSession}
+     */
```

Comment:
comment should explain what sessionIndex is

Reply:
Sure! Will add that in.